src/ga/operators/crossover.rs

```diff
@@ -356,3 +358,100 @@ where
     (child_1, child_2)
   }
 }
+
+/// # Ordered crossover operator
+///
+/// This struct implements [self::CrossoverOperator] trait and can be used with GA.
+///
+/// It works by taking a substring from one parent, and filling the missing places with alleles from
+/// second parent in the order they appear in.
+///
+/// P1 : 2 <b>4 1 3</b> 5 <br>
+/// P2 : 5 2 1 4 3 <br>
+/// Ch : 5 <b>4 1 3</b> 2
+///
+/// Degenerated case when substring has length equal to genome length can occur.
+pub struct OrderedCrossover;
+
+impl OrderedCrossover {
+  /// Creates new [self::SinglePoint] crossover operator
+  pub fn new() -> Self {
+    OrderedCrossover {}
+  }
+}
+
+impl<GeneT, ChT> CrossoverOperator<ChT> for OrderedCrossover
+where
+  ChT: Chromosome + Index<usize, Output = GeneT> + Push<GeneT, PushedOut = Nothing>,
+  GeneT: Copy + Eq + Hash,
+{
+  /// Returns a tuple of children, first child is created by taking a substring from parent_1,
+  /// second child is created by using a substring from parent_2
+  ///
+  /// It works by taking a substring from one parent, and filling the missing places with alleles from
+  /// second parent in the order they appear in.
+  ///
+  /// P1 : 2 <b>4 1 3</b> 5 <br>
+  /// P2 : 5 2 1 4 3 <br>
+  /// Ch : 5 <b>4 1 3</b> 2
+  ///
+  /// Degenerated case when substring has length equal to genome length can occur.
+  ///
+  /// ## Arguments
+  ///
+  /// * `parent_1` - First parent to take part in crossover
+  /// * `parent_2` - Second parent to take part in crossover
+  fn apply(
+    &mut self,
+    parent_1: &Individual<ChT>,
+    parent_2: &Individual<ChT>,
+  ) -> (Individual<ChT>, Individual<ChT>) {
+    assert_eq!(
+      parent_1.chromosome_ref().len(),
+      parent_2.chromosome_ref().len(),
+      "Parent chromosome length must match"
+    );
+
+    let chromosome_len = parent_1.chromosome_ref().len();
+
+    let begin: usize = rand::thread_rng().gen_range(0..chromosome_len);
+    let end: usize = begin + rand::thread_rng().gen_range(0..=(chromosome_len - begin));
+
+    let create_child = |p1: &Individual<ChT>, p2: &Individual<ChT>| -> Individual<ChT> {
+      let mut substring_set: HashSet<GeneT> = HashSet::new();
+
+      for i in begin..end {
+        substring_set.push(p1.chromosome_ref()[i]);
+      }
+
+      let mut child: Individual<ChT> = Individual::new();
+      let mut index: usize = 0;
+
+      while child.chromosome_ref().len() < begin {
+        let gene = p2.chromosome_ref()[index];
+        if !substring_set.contains(&gene) {
+          child.chromosome_ref_mut().push(gene);
+        }
+        index += 1;
+      }
+
+      for i in begin..end {
+        child.chromosome_ref_mut().push(p1.chromosome_ref()[i]);
+      }
+
+      while index < chromosome_len {
+        let gene = p2.chromosome_ref()[index];
+        if !substring_set.contains(&gene) {
+          child.chromosome_ref_mut().push(gene);
+        }
+        index += 1;
+      }
+      child
+    };
```

Comment:
I would refactor it out to separate helper function (private module function) as using closure here does not look readable to me (just my personal opinion).